Local frame TwistStamped output from diff_drive plugin

robustify opened the pull request:
PR to implement proposed feature from #40 

I used the same update timer that already exists, so the twist output will publish at the rate of the 'pub_rate' parameter.  In some of my simulation scenarios, I don't want there to be odometry topics available, but in others I would like odometry but not local frame twist.  Therefore, I added two boolean parameters 'pub_odom' and 'pub_twist' that enables advertising of each type of output topic.

However, now there are confusing parameter names. There is 'odom_pub', which is the topic name of the noisy odometry output, and 'pub_odom', which enables this topic.  Same issue between 'twist_pub' and 'pub_twist'. I would change 'odom_pub' to either 'odom_topic' or 'odom_pub_topic', but I'll leave that up to you.

josephduchesne commented on flatland_plugins/src/diff_drive.cpp:
```diff
@@ -61,6 +61,8 @@ void DiffDrive::TwistCallback(const geometry_msgs::Twist& msg) {
 
 void DiffDrive::OnInitialize(const YAML::Node& config) {
   YamlReader reader(config);
+  pub_odom_ = reader.Get<bool>("pub_odom", true);
```
maybe "enable_odom_pub" and "enable_twist_pub" for these to reduce confusion with the existing variable names.

robustify commented:
Changed parameter names and updated documentation!  Should I make the local frame twist output default to being enabled?  I made it default to false to not change the behavior of any existing simulation setups.

josephduchesne commented:
It probably won't break anything to enable twist output by default.  If you want to check, you can try the [turtlebot flatland](https://github.com/avidbots/turtlebot_flatland) package with your changes. I don't currently have any other projects that use the differential drive plugin.

robustify commented:
It works fine with my simulations using diff_drive. I just wanted to make sure it wouldn't be an issue for you having to set the enable_twist_pub parameter to false explicitly in all your existing models if you didn't want the twist output. Since it is OK on your end, I'll make it default to true.
